src/datasource.js

```diff
@@ -30,24 +30,16 @@ export class GenericDatasource {
       query.adhocFilters = [];
     }
 
+    // extract variable definition from index and add to scopedVars
     const index = _.isUndefined(this.templateSrv.index) ? {} : this.templateSrv.index;
-    const variables = {};
-    Object.keys(index).forEach(function (key) {
-      const variable = index[key];
-      variables[variable.name] = {
-        text: variable.current.text,
-        value: variable.current.value
+    _.assign(options.scopedVars, ... _.map(index, (val, key) => {
```

Comment:
I think this is actually easier to read and understand than the original version even though its using lodash.